### PR TITLE
build: POM configured to utilize Java-17 compiler features. 

### LINKED ISSUE #431: Upgrade Scene Builder Java compiler version to Java 17
Hi,
I think upgrading the Scene Builder Java compiler options in `POM.xml` can be helpful for future work. 
First, Java-17 is the new LTS version and it brings a number of enhancements and language features.
Therefore, I thing upgrading Java compiler version in `POM.xml` from 11 to 17 is a good idea.

How do you think about @abhinayagarwal & @jperedadnr ?

### Expected Behavior

Project uses Java-17 language features.

```xml
    <properties>
        <javafx.version>17</javafx.version>
        <maven.compiler.release>17</maven.compiler.release>
    </properties>
```

### Current Behavior

Project uses Java-11 language features.
```xml
    <properties>
        <javafx.version>17</javafx.version>
        <maven.compiler.release>11</maven.compiler.release>
    </properties>
```

### Context

Java-17 is the new LTS version of Java. As Scene Builder is shipped with a Java-17 runtime, it could also be helpful to build it using Java-17 language features such as records, text blocks or better pattern matching.

### PR DESCRIPTION
**Pull request by Oliver-Loeffler**
Scene Builder POM configured to make use of most recent LTS Java version 17 language features.
Readme updated with reference to JDK17 accordingly.

### Issue

Fixes #431

### Progress

- [x] Change must not contain extraneous whitespace
- [x] License header year is updated, if required
- [x] Verify the contributor has signed [Gluon Individual Contributor License Agreement (CLA)](https://docs.google.com/forms/d/16aoFTmzs8lZTfiyrEm8YgMqMYaGQl0J8wA0VJE2LCCY)

### CODE REVIEW

**Comment by Oliver-Loeffler** [comment]
After some reading and reviewing how Kit is used, I think that kit should remain on Java-11 compiler level.
For app, the Java-17 additions can really help, therefore I limited the Java-17 compiler level change to the app module.

File: app/pom.xml [hunk 1]
```diff
@@ -17,6 +17,7 @@
         <javaVersion>${java.version}, ${java.runtime.name}</javaVersion>
         <javafxVersion>${javafx.version}</javafxVersion>
         <maven.build.timestamp.format>yyyy-MM-dd HH:mm:ss</maven.build.timestamp.format>
+        <maven.compiler.release>11</maven.compiler.release>
```
Comment (abhinayagarwal) [inline app/pom.xml:20]: s/11/17 ?

**Comment by abhinayagarwal** [comment]
I was of the opinion that we want the project to still build on 11 and just the app on 17. With recent changes, the entire project is moved to 17.

**Comment by Oliver-Loeffler** [comment]
I have corrected it accordingly, kit is now back to 11 whereas app is back to 17.

File: app/pom.xml [hunk 2]
```diff
@@ -14,7 +14,7 @@
         <main.class.name>com.oracle.javafx.scenebuilder.app.SceneBuilderApp</main.class.name>
         <!-- For about.properties file -->
         <buildDate>${maven.build.timestamp}</buildDate>
-        <javaVersion>${java.version}, ${java.runtime.name}</javaVersion>
+        <javaVersion>${java.version.app}, ${java.runtime.name}</javaVersion>
```
Comment (abhinayagarwal) [inline app/pom.xml:17]: Isn't this property only used to show details in the application and doesn't have any effect on how the `app` project is built via Maven.

If you want `app` to be built using Java 17, `<java.version>17</java.version>` needs to be added to `app/pom.xml` properties section.

Comment (Oliver-Loeffler) [reply app/pom.xml:17]: This is indeed a mistake. There is now a property `<java.version>${java.version.app}</java.version>` and it runs.

**Comment by AlmasB** [comment]
Closed by #527

**Pull request closed without merging**